tests/by-util/test_rev.rs

```diff
@@ -24,6 +24,18 @@ fn test_existing_file() {
     ucmd.arg("a.txt").succeeds().stdout_is("A enil\nB enil");
 }
 
+#[test]
+fn test_zero() {
+    let (at, mut ucmd) = at_and_ucmd!();
+
+    at.write("a.txt", "line A\nline B");
+
+    ucmd.arg("a.txt")
+        .arg("--zero")
+        .succeeds()
+        .stdout_is("A enil\0B enil");
```

Comment:
I think there is a misunderstanding of how `rev  --zero` works (and I should have added an example to the ticket).

With `--zero`, `rev` uses `\0` as a line separator in the _input_.

```
// input is processed as a single line because there is no \0
$ printf "line A\nline B" | rev --zero | cat -A
B enil$
A enil

// input is processed as two lines
$ printf "line A\0line B" | rev --zero | cat -A
A enil^@B enil
```

Reply:
fixed!